### ISSUE TITLE
19.09 + lightdm + home-manager cannot log in

### ISSUE THREAD

**Issue by Ericson2314** [NixOS/nixpkgs#70142]
**Describe the bug**

When I tried 19.09 (https://github.com/Ericson2314/nixos-configuration/commit/4182b732eb1b07cc6fa0070108e75ffa9584a531 should be representative, I am booting from an old nixos generation with that ATM), LightDM no longer had me under the easy-access dropdown list of users. When I logged in by typing my username instead, the login failed, and I couldn't find anything fresh in:

1. `~/.xsesssion-errors`
2. `/var/log/lightdm.log`
3 `journalctl --user <stuff>`

I temporary switched display managers to slim, and that did work. (well my user systemd units seemed a bit messed up, but I am not sure that is releted.)

**To Reproduce**
1. NixOS 19.09
2. lightdm as desktop mananger
3. home manager controlled xsession
4. Try to log in

**Expected behavior**

**Screenshots**

**Additional context**

@matthewbauer thought that perhaps there was some dbus thing was down, and so lightdm neither learned who the list of users was, nor had enough information do the log in.

**Metadata**
Please run `nix run nixpkgs.nix-info -c nix-info -m` and paste the result.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by worldofpeace** [comment]
Yeah, the only thing that could be "some dbus thing was done" would probably be accountsservice.

Anything funny going in the files at `/var/lib/AccountsService/users`, for example `SystemAccount=True`?

**Comment by worldofpeace** [comment]
> home manager controlled xsession

I believe we've disabled xterm in 19.09 in some manner, so perhaps try enabling `services.xserver.desktopManager.xterm.enable`. IIRC home-manager was somehow relying on that.

**Comment by Ericson2314** [comment]
@worldofpeace Yup! `services.xserver.desktopManager.xterm.enable = true` fixed it. Thanks! @rycee heads up for installation instructions etc.

Now that just leaves eh missing user list, which I should open a separate issue for.

**Comment by Ericson2314** [comment]
Regardless of home manager, there probably should be some sort of state dir or warning or something that xterm isn't on by default?

**Comment by Ralith** [comment]
It looks like exactly this same issue has happened again in 20.03, which resulted in an unpleasant half-hour of vterm configuration debugging when I updated. lightdm requires some sessions defined to let you log in, and there are none defined by default. Enabling any session at all fixes things, because the nixos session wrapper script tries ~/.xsession before doing what it's supposed to.

I see https://github.com/NixOS/nixpkgs/pull/68371 might have been a fix, but it's been abandoned for a few months.

**Comment by worldofpeace** [comment]
@Ralith The solution is to enable some sort of desktop manager, xterm. It was in the release notes when this was relevant. The bug is more of a home-manager issue, so it lives there https://github.com/rycee/home-manager/issues/1116. The docs could be updated there to enable xterm, or it could provide a nixos module with an empty one.

**Comment by Ralith** [comment]
> It was in the release notes when this was relevant.

This was not in the 20.03 release notes, and it worked fine before updating, presumably due to the above reversions. Indeed carefully reviewed the 20.03 release notes before updating, and this did not prevent my system from being broken.

**Comment by worldofpeace** [comment]
@Ralith What I meant by "when this was relevant" is the issue existed in 19.09. It's indeed interesting you'd encounter this now though, did you change your stateVersion?

**Comment by Ralith** [comment]
No, my stateVersion remains at 19.09.

**Comment by rycee** [comment]
@Ralith An alternative might be to do something like the suggestion here: https://discourse.nixos.org/t/opening-i3-from-home-manager-automatically/4849/8
